### PR TITLE
Fix long bundle names

### LINKED ISSUE #523: fleet-agent cleanup continually deleting releases
The cleanup loop in `fleet-agent` isn't able to properly identify releases with bundles. For a given release, it looks for the bundle deployment associated to that release. If the release has a different name than fleet-agent expects (based on the bundle deployment) then fleet-agent deletes the release.

The problem is that then a release with the exact same name is created and the next time the cleanup loop runs, fleet-agent will delete the release again.

For example, a bundle deployment with name `mcc-anupamafinalrcrke2-managed-system-upgrade` creates a release with name `mcc-anupamafinalrcrke2-managed-system-upgrade-c-407d2`. 

It seems that [this function](https://github.com/rancher/fleet/blob/master/modules/agent/pkg/deployer/manager.go#L42) should look at `status.release` as well when trying to determine the name of the release from a bundle deployment.

### PR DESCRIPTION
**Pull request by manno**
Fix #523

The bundle name is used in several places:
* pruning bundles when running "fleet apply"
* finding bundles when fleet controller runs CleanUp
* as a default for the helm release name
* as the name of the bundledeployment (in the clusters namespace)

The bundle name usually consists of the "gitrepo" name and the path in which the bundle was found.

Helm allows for release names of up to 53 characters. When creating the bundle name from git "repoName" + "path to bundle", fleet now limits the lenght to 53. It will also replace all disallowed characters.

If the user specifies a custom release name in the helm options (via fleet.yaml), that will be validated, too. JSON schema was added to the CRD.

### Tradeoff

Old bundle names should remain stable, so they are not reinstalled. Unless the old names are too long, however in that case they are in danger of a collision or being falsely targeted by fleet controller's cleanup.

### CODE REVIEW

File: modules/cli/apply/apply.go [hunk 1]
```diff
@@ -167,10 +174,46 @@ func readBundle(ctx context.Context, name, baseDir string, opts *Options) (*bund
...
+	}
+
+	// clean up new name
+	str = strings.ToLower(str)
+	str = disallowedChars.ReplaceAllLiteralString(str, "-")
+	str = multiDash.ReplaceAllString(str, "-")
+	str = strings.TrimLeft(str, "-")
+	str = strings.TrimRight(str, "-")
+
+	// shorten name to 53 characters, the limit for helm release names
```
Comment (thardeck) [inline modules/cli/apply/apply.go:208]: I would extend the comment that the Limit function of wrangler adds a checksum to prevent collisions itself.

**Review by thardeck — APPROVED** [review]
LGTM besides the nitpick, but it is not mandatory.

**Comment by thardeck** [comment]
And thanks for the many additional comments to improve code readability. :bouquet:

**Comment by manno** [comment]
I think this also fixes https://github.com/rancher/fleet/issues/196, https://github.com/rancher/fleet/issues/911 and https://github.com/rancher/fleet/issues/156

**Pull request merged**